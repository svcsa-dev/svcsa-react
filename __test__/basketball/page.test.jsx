/**
 * @jest-environment jsdom
 */
import { render, screen } from "@testing-library/react";
import Page from "@/app/basketball/page";
import { beforeEach } from "node:test";
import fetchMock from "jest-fetch-mock";

describe("Home", () => {
  beforeEach(() => {
    fetchMock.resetMocks();
  });

  it("renders the basic page", async () => {
    const mockData = {
      total: 12,
      limit: 3,
      skip: 0,
      data: [
        {
          id: 18,
          name: "男篮公开组第21赛季",
          starttime: "2022-09-18T00:00:00.000Z",
          teamnumber: 20,
          groupnumber: 1,
          playoffgroupnumber: 1,
          rules: 3,
          competitionid: 1,
          startdate: {
            month: 9,
            day: 18,
            year: 2022,
          },
          competition: {
            id: 1,
            name: "AlphaX杯SVCBA男篮公开组",
            description: "硅谷华人男篮联赛公开组",
            picture:
              "http://svcsa.org/uploads/20220415/deeb9c494c1d6cd2cf682fe978e36a02.jpg",
          },
        },
        {
          id: 19,
          name: "女篮公开组第7赛季",
          starttime: "2022-09-18T00:00:00.000Z",
          teamnumber: 5,
          groupnumber: 1,
          playoffgroupnumber: 1,
          rules: 2,
          competitionid: 2,
          startdate: {
            month: 9,
            day: 18,
            year: 2022,
          },
          competition: {
            id: 2,
            name: "AlphaX杯SVCBA女篮公开组",
            description: "硅谷华人篮球联赛女篮公开组",
            picture:
              "http://svcsa.org/uploads/20220415/a0be11abe00633a950a7a9c8d651df5e.jpg",
          },
        },
        {
          id: 20,
          name: "男篮壮年组第1赛季",
          starttime: "2022-09-18T00:00:00.000Z",
          teamnumber: 4,
          groupnumber: 1,
          playoffgroupnumber: 1,
          rules: 2,
          competitionid: 4,
          startdate: {
            month: 9,
            day: 18,
            year: 2022,
          },
          competition: {
            id: 4,
            name: "AlphaX杯SVCBA男篮壮年组",
            description: "硅谷华人篮球联赛男篮壮年组",
            picture:
              "http://svcsa.org/uploads/20220816/829ecc8922c42936100e634e71716684.jpg",
          },
        },
      ],
    };

    // Mock the fetch response
    fetchMock.mockResponseOnce(JSON.stringify(mockData));
    const serverComponent = await Page();
    const {container} = render(serverComponent);

    const heading = screen.getByText("Hello, basketball league page!");

    expect(heading).toBeInTheDocument();

    const standingLinks = screen.getAllByTestId("standing-link");

    expect(standingLinks.length).toBe(3);

    expect(container).toMatchSnapshot();
  });
});
