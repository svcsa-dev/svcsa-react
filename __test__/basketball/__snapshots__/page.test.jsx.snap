// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders the basic page 1`] = `
<div>
  <section>
    <h1>
      Hello, basketball league page!
    </h1>
    <section>
      <h3>
        Standings 
      </h3>
      <a
        class="relative items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-primary no-underline hover:opacity-80 active:opacity-disabled transition-opacity block"
        data-testid="standing-link"
        href="/basketball/men-open/standing"
        role="link"
        tabindex="0"
      >
        男篮2024-2025赛季
      </a>
      <a
        class="relative items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-primary no-underline hover:opacity-80 active:opacity-disabled transition-opacity block"
        data-testid="standing-link"
        href="/basketball/women-open/standing"
        role="link"
        tabindex="0"
      >
        女篮2024-2025赛季
      </a>
    </section>
    <br />
    <section>
      <h3>
        Teams
      </h3>
      <a
        class="relative items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-primary no-underline hover:opacity-80 active:opacity-disabled transition-opacity block"
        data-testid="teams-link"
        href="/basketball/men-open/teams"
        role="link"
        tabindex="0"
      >
        男篮2024-2025赛季
      </a>
      <a
        class="relative items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-primary no-underline hover:opacity-80 active:opacity-disabled transition-opacity block"
        data-testid="teams-link"
        href="/basketball/women-open/teams"
        role="link"
        tabindex="0"
      >
        女篮2024-2025赛季
      </a>
    </section>
    <br />
    <section>
      <div
        class="grid grid-cols-5 bg-slate-950 h-72"
      >
        <div
          class=" col-span-2 text-white self-center items-center grid justify-items-center mt-7"
        >
          <h3
            class="font-semibold text-3xl "
          >
            News At 
            <span
              class="text-orange-700"
            >
              SVCSA
            </span>
          </h3>
          <button
            class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-unit-4 min-w-unit-20 h-unit-10 text-small gap-unit-2 rounded-medium [&>svg]:max-w-[theme(spacing.unit-8)] data-[pressed=true]:scale-[0.97] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover bg-orange-700 text-white mt-3 cursor-pointer"
            type="button"
          >
            More news
          </button>
        </div>
        <p>
          Loading...
        </p>
      </div>
    </section>
  </section>
</div>
`;
